build-tools/src/main/java/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -94,10 +122,59 @@ public void setSpoolOutput(boolean spoolOutput) {
             };
         } else {
             out = new ByteArrayOutputStream();
-            outputLogger = logger -> { logger.error(((ByteArrayOutputStream) out).toString(StandardCharsets.UTF_8)); };
+            outputLogger = getIndentingConsoleOutput().isPresent() ? logger -> {} : logger -> logger.error(byteStreamToString(out));
+        }
+
+        OutputStream finalOutputStream = getIndentingConsoleOutput().isPresent()
+            ? new IndentingOutputStream(System.out, getIndentingConsoleOutput().get())
+            : out;
+        ExecResult execResult = execOperations.exec(execSpec -> {
+            execSpec.setIgnoreExitValue(true);
+            execSpec.setStandardOutput(finalOutputStream);
+            execSpec.setErrorOutput(finalOutputStream);
+            execSpec.setExecutable(getExecutable().get());
+            execSpec.setEnvironment(getEnvironment().get());
+            if (getArgs().isPresent()) {
+                execSpec.setArgs(getArgs().get());
+            }
+            if (getWorkingDir().isPresent()) {
+                execSpec.setWorkingDir(getWorkingDir().get());
+            }
+            if (getStandardInput().isPresent()) {
+                try {
+                    execSpec.setStandardInput(new ByteArrayInputStream(getStandardInput().get().getBytes("UTF-8")));
+                } catch (UnsupportedEncodingException e) {
+                    throw new GradleException("Cannot set standard input", e);
+                }
+            }
+        });
+        int exitValue = execResult.getExitValue();
+
+        if (exitValue == 0 && getCaptureOutput().get()) {
+            output = byteStreamToString(out);
+        }
+        if (getLogger().isInfoEnabled() == false) {
+            if (exitValue != 0) {
+                try {
+                    getLogger().error("Output for " + getExecutable().get() + ":");
```

Comment:
I think we want to omit this is indenting output is enabled so we don't print `Output for foo:` and then nothing else.